zjit/src/hir.rs

```diff
@@ -138,6 +138,40 @@ impl Invariant {
     }
 }
 
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub enum SpecialObjectType {
+    VMCore = 1,
+    CBase = 2,
+    ConstBase = 3,
+}
+
+impl From<u32> for SpecialObjectType {
+    fn from(value: u32) -> Self {
+        match value {
+            1 => SpecialObjectType::VMCore,
```

Comment:
Actually, since `rb_mRubyVMFrozenCore` is used expressly by YJIT already, we should be able to do the same. So please instead compile the `1` case into a `Const` instruction with `rb_mRubyVMFrozenCore`